Feature/chapter 11 supabase

eightsenses opened the pull request.

eightsenses committed:
.envファイルを.gitignoreに追加

eightsenses committed:
feat: update dependencies and integrate Supabase for authentication

- Updated @prisma/client and prisma to version 6.11.1
- Added @supabase/supabase-js for Supabase integration
- Refactored Post model to use thumbnailImageKey instead of thumbnailUrl
- Implemented useSupabaseSession hook for managing user session
- Enhanced Header component to include login/logout functionality based on session state
- Created login and signup pages for user authentication
- Updated post API to handle authorization tokens
- Added route guard for admin pages to restrict access to authenticated users
- Modified post creation and editing forms to handle image uploads to Supabase storage
- Updated frontend components to reflect changes in post data structure

eightsenses committed:
fix: redirect to home after logout instead of login

eightsenses committed:
fix: update chapter titles for layout metadata

eightsenses committed:
Remove .env from tracking

eightsenses committed:
feat: implement AuthForm component and refactor login/signup pages to use it

shuheitachikawa commented on src/app/admin/_hooks/useRouteGuard.ts:
```diff
@@ -0,0 +1,20 @@
+  useEffect(() => {
+    if (session === undefined) return; // sessionがundefinedの場合は読み込み中なので何もしない
+
+    const fetcher = async () => {
+      if (session === null) {
+        router.replace('/login');
+      }
+    };
```
非同期処理が含まれていないので、関数定義不要で、13~15行目を実行するだけで良さそうです。

shuheitachikawa replied:
こちら、まだ修正されていなそうですかね、？

shuheitachikawa replied:
```tsx
useEffect(() => {
  // まだセッション情報を取得中
  if (session === undefined) return;

  // 未ログインならログインページへリダイレクト
  if (session === null) {
    router.replace('/login');
  }
}, [session, router]);
```
こんな感じになるかと！

eightsenses replied:
ありがとうございます!!
対応済みです!!
useEffectに移動しました。

対象行を変更したため自動で反映されなかった??ですかね??

shuheitachikawa commented on src/app/api/admin/posts/[id]/route.ts:
api/admin以下のroute.tsは、tokenがないと実行できないようにする必要があるので、すべての関数に認可の処理を入れましょう。

shuheitachikawa replied:
api/admin/categories内も、対応必要そうです！

shuheitachikawa reviewed (commented):
コメントいたしました！

あと、今くらいのタイミングで、
データフェッチ＋State管理はすべてSWR、
フォームのあたり管理はすべてreact-hook-form
のライブラリに置き換えてみましょう！
それぞれ、広く使われていて、覚えておくととても便利です。
https://swr.vercel.app/ja
https://react-hook-form.com/

eightsenses committed:
fix: update dev script to bind to all network interfaces

eightsenses committed:
fix: simplify session check in useRouteGuard hook

eightsenses committed:
feat: add token-based authentication to category and post APIs, update related components

eightsenses committed:
feat: add SWR library and update brace-expansion versions in package-lock.json

eightsenses committed:
fix: correct dev script host binding and add SWR dependency

eightsenses committed:
Refactor admin category and post management with SWR for data fetching

- Replaced manual data fetching with SWR in CategoryList and AdminPostsList components.
- Simplified state management in EditPost and CreatePost components by utilizing custom hooks.
- Updated PostForm component to use react-hook-form for form handling and validation.
- Improved error handling and loading states across various components.
- Removed unused code and types related to the contact form.
- Enhanced user experience with appropriate status messages during data fetching and submission.

shuheitachikawa reviewed (commented):
１点、コメントいたしました！

shuheitachikawa reviewed (approved):
LGTMです！！
